### DISCUSSION TITLE
about SelectionScopeField，Is there a bug? In iTwin.js 3.1.0-dev.6 version

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#3076] [Ideas] [answered]
Hi，I'm using a SelectionScopeField and found that there is only one item "element" in the Select list, "Assembly" and "Top Assembly" are missing. 
![1](https://user-images.githubusercontent.com/26344907/150522852-7c5aa730-ec8e-48cd-a2bd-7b3d57fb334c.png)
Then, I tried to take out the source file SelectionScope.tsx to modify it and then use the display and function to work correctly. 
![image](https://user-images.githubusercontent.com/26344907/150523310-8161c373-38c3-401b-a7ba-f7d1a908b7d2.png)
My simple modification is as follows: 
![image](https://user-images.githubusercontent.com/26344907/150523556-095ee865-45cf-49f5-9cfe-f9c3bd4a3558.png)
So, am I not using SelectionScopeField correctly? Or is there really a bug?
My same code was correct in itwin.js 2.19.27, upgraded to itwin.js 3.1.0-dev.6 version and found this problem. thanks.

**Comment by grigasp** [comment]
As far as I can tell, the component takes the scopes' list from `UiFramework.frameworkState.sessionState.availableSelectionScopes` ([here](https://github.com/iTwin/itwinjs-core/blob/a9a7a03c5281478370afb9f66094bf0e56c5e883/ui/appui-react/src/appui-react/statusfields/SelectionScope.tsx#L89)). And it looks like it returns the default value (based on [this](https://github.com/iTwin/itwinjs-core/blob/a9a7a03c5281478370afb9f66094bf0e56c5e883/ui/appui-react/src/appui-react/redux/SessionState.ts#L74)), which only contains the "element" scope. The comment says it's going to be overwritten when iModelConnection is established, but I can't see where that's happening. @bsteinbk?

**Comment by bsteinbk** [comment, marked as answer]
@shenshengyi - You should be calling **[UiFramework.setIModelConnection](https://github.com/iTwin/itwinjs-core/blob/master/ui/appui-react/src/appui-react/UiFramework.ts)** each time you open a new IModel.  In 3.0 this method will then call [SyncUiEventDispatcher.initializeConnectionEvents](https://github.com/iTwin/itwinjs-core/blob/master/ui/appui-react/src/appui-react/syncui/SyncUiEventDispatcher.ts) which internally calls Presentation.selection.scopes.getSelectionScopes and updates the Redux store triggering an event which should cause the UI to refresh.

**Reply by shenshengyi** [reply]
hi, I have called UiFramework.setIModelConnection , which is also written in itwin.js 19.27. Currently it doesn't display correctly in itwin.js 3.
My code is as follows: 
![image](https://user-images.githubusercontent.com/26344907/150618146-47dfe014-7668-401a-8fba-4c3d9aeb0018.png)
@bsteinbk Thanks.

**Reply by bsteinbk** [reply]
@shenshengyi  - You don't need to call SyncUiEventDispatcher.initializeConnectionEvents any more on line 66 as that is now being called within UiFrameworksetIModelConnection. The only thing two things that I see would not properly load the scopes is if iModelConnection.isBlankConnection() returns true or the call Presentation.selection.scopes.getSelectionScopes does not retun any scopes.  Maybe you can try debugging through the method [SyncUiEventDispatcher.initializeConnectionEvents ](https://github.com/iTwin/itwinjs-core/blob/master/ui/appui-react/src/appui-react/syncui/SyncUiEventDispatcher.ts).

**Reply by shenshengyi** [reply]
OK,Thanks.
